From df13aa2820872a79816b40eec124dc6e50e8544a Mon Sep 17 00:00:00 2001
From: Javier Celaya <javier.celaya@flexvdi.com>
Date: Tue, 18 Apr 2017 10:51:52 +0200
Subject: [PATCH 2/5] Set errno to EWOULDBLOCK when there is no full message

In nopoll_conn_read, if nopoll_conn_get_msg returns NULL, the connection
is still ok and a non-blocking call has been requested, return
EWOULDBLOCK
---
 src/nopoll_conn.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/nopoll_conn.c b/src/nopoll_conn.c
index c8ebd28..20ba8fa 100644
--- a/src/nopoll_conn.c
+++ b/src/nopoll_conn.c
@@ -3773,8 +3773,14 @@ int           nopoll_conn_read (noPollConn * conn, char * buffer, int bytes, nop
 			} /* end if */
 
 			if (! block) {
-				if (total_read == 0 && ! block) 
+				if (total_read == 0 && ! block) {
+#if defined(NOPOLL_OS_UNIX)
+					errno = NOPOLL_EWOULDBLOCK; /* simulate there is no data available */
+#elif defined(NOPOLL_OS_WIN32)
+					WSASetLastError(NOPOLL_EWOULDBLOCK); /* simulate there is no data available */
+#endif
 					return -1;
+                                }
 				return total_read;
 			} /* end if */
 			
-- 
2.9.3

